src/platform/Linux/CHIPBluezHelper.h

```diff
@@ -205,6 +205,63 @@ CHIP_ERROR StopDiscovery(BluezEndpoint * apEndpoint);
 
 CHIP_ERROR ConnectDevice(BluezDevice1 * apDevice);
 
+/// Iterates over available BlueZ adapters
+///
+/// Usage example:
+///
+///  AdapterIterator iterator;
+///  while (iteerator.Next()) {
+///      std::cout << iterator.GetAddress() << std::endl;
+///  }
+class AdapterIterator
+{
+public:
+    ~AdapterIterator();
+
+    /// Moves to the next DBUS interface.
+    ///
+    /// MUST be called before any of the 'current value' methods are
+    /// being used (iterator gets initialized on the first call of Next).
+    bool Next();
+
+    // Information about the current value. Save to call only after
+    // "Next" has returned true.
+    uint32_t GetIndex() const { return mCurrent.index; }
+    const char * GetAddress() const { return mCurrent.address; }
+    const char * GetAlias() const { return mCurrent.alias; }
+    const char * GetName() const { return mCurrent.name; }
+    bool IsPowered() const { return mCurrent.powered; }
+
+private:
+    /// Sets up the DBUS manager and loads the list
+    void Initialize();
+
+    /// Loads the next value in the list.
+    ///
+    /// Returns true if a value could be loaded, false if no more items to
+    /// iterate through.
+    bool Advance();
+
+    static constexpr size_t kMaxAddressLength = 19; // xx:xx:xx:xx:xx:xx
+    static constexpr size_t kMaxNameLength    = 64;
+
+    GDBusObjectManager * mManager = nullptr; // DBus connection
+    GList * mObjectList           = nullptr; // listing of objects on the bus
+    GList * mCurrentListItem      = nullptr; // current item viewed in the list
+
+    bool mHasCurrentValue = false;
+
+    // data valid only if Next() returns true
+    struct
+    {
+        uint32_t index;
+        char address[kMaxAddressLength];
```

Comment:
std::string?

I'd posit that we're wasting our time when we try to apply the same constraints we have on tightly constrained code to the Linux port.

Reply:
Agree. I have a hard time figuring out when I should/should not use constrained stuff so I default to uglier code.

I would prefer to leave as is for this PR because less changes and CI takes a long time. Will take it into account if I touch this later.